### ISSUE TITLE
bug: action fails to setup due to `--user` in embedded python for windows

### ISSUE THREAD

**Issue by Ravio1i** [EnricoMi/publish-unit-test-result-action#562]
## Description

When using the composite version of this action on windows, where embedded python version is used, the following error is shown:

```bash
Install Python dependencies.
  ******/test-composite-action/enricomi-publish-action-venv/Scripts/python3
  ERROR: Can not perform a '--user' install. User site-packages are not visible in this virtualenv.
  Error: Process completed with exit code 1.
```

This is happening due to the always enforced `pip_option=--user` for Windows [here](https://github.com/enricomi/publish-unit-test-result-action/blob/master/composite/action.yml#L193). This was introduced with this `virtualenv` [commit](https://github.com/enricomi/publish-unit-test-result-action/commit/78b628159a70e2c81b88e2e50914f1b00fa60a42).

The action actually runs fine, once we remove the `--user` part from it.

Would it be possible to make the `--user` option for windows optional. Obviously, it can be the default. 

```yaml
inputs:
  pip_user_option:
    description: ' Use --user option for pip'
    required: false
    default: ' false'

- name: Detect OS
  id: os
  run: |
    case "$RUNNER_OS" in
      Linux*)
        echo "pip-cache=~/.cache/pip" >> $GITHUB_OUTPUT
        ;;
      macOS*)
        echo "pip-cache=~/Library/Caches/pip" >> $GITHUB_OUTPUT
        ;;
      Windows*)
        echo "pip-cache=~\\AppData\\Local\\pip\\Cache" >> $GITHUB_OUTPUT
        if [ "${{ inputs.pip_user_option }}" == "true" ]; then
          echo "pip-options=--user" >> $GITHUB_OUTPUT
        fi
        ;;
    esac
  shell: bash
```

**Comment by EnricoMi** [comment]
@Ravio1i please test my fix branch:

    uses: EnricoMi/publish-unit-test-result-action/composite@check-venv-installed

If you have the `virtualenv` or `venv` package installed, it does not call `pip install` at all. If they are not installed, it first tries `pip install --user` and when that fails it tries without `--user`. So that should work for you in any case.

**Comment by EnricoMi** [comment]
And I removed the `--user` option from `Install Python dependencies` because I think that was wrong in the first place.

**Comment by Ravio1i** [comment]
@EnricoMi Using the suggested branch works! Thank you!

When can we expect it to be within the new release of this action?

**Comment by EnricoMi** [comment]
Fixed in #558.

**Comment by EnricoMi** [comment]
This has just been released.
